File: src/main/java/com/gclasscn/base/security/SecurityUser.java
package com.gclasscn.base.security;

import java.util.Collection;

import org.springframework.security.core.GrantedAuthority;

import com.gclasscn.base.domain.User;


public class SecurityUser extends org.springframework.security.core.userdetails.User{

	private static final long serialVersionUID = 4231489117252871269L;
	
	private User user;
	
	public SecurityUser(User user, Collection<? extends GrantedAuthority> authorities) {
		super(user.getAccount(), user.getPassword(), true, true, true, true, authorities);
		this.user = user;
	}

	public User getUser() {
		return user;
	}

	public void setUser(User user) {
		this.user = user;
	}
	
}

File: src/main/java/com/gclasscn/base/security/SecurityUser.kt
package com.gclasscn.base.security;

import org.springframework.security.core.GrantedAuthority;

import com.gclasscn.base.domain.User;


class SecurityUser : org.springframework.security.core.userdetails.User{

	var user: User? = null;
	
	constructor(user: User, authorities: MutableCollection<GrantedAuthority>):
			super(user.account, user.password, true, true, true, true, authorities) {
		this.user = user;	
	}
	
}
